dlib/pixel.h

```diff
@@ -107,6 +107,43 @@ namespace dlib
                 - is_unsigned is true if min() == 0 and false otherwise
     !*/
 
+// ----------------------------------------------------------------------------------------
+
+    namespace details
+    {
+        template<typename...T>
+        using void_t = void;
```

Comment:
This is getting used in multiple files here an I'm sure will end up in other places.  It should live somewhere more common.  There are a bunch of similar one liners like this in dlib/algs.h.  So put it there, and I think it's fine if it's just `dlib::void_t`.  Then things can just use it.  Doubtless it will appear in things unrelated to pixels eventually :)

Reply:
It's already in `dlib/type_traits.h` as a backport of C++17's `void_t` which also lives in `<type_traits>`.
So I had already done the right thing it seems. I'll just remove the duplicates and just `#include <dlib/type_traits.h>`